OpenRA.Game/Map/Map.cs

```diff
@@ -719,11 +713,34 @@ public byte[] SavePreview()
 
 			var isRectangularIsometric = Grid.Type == MapGridType.RectangularIsometric;
 
-			// Fudge the heightmap offset by adding as much extra as we need / can.
-			// This tries to correct for our incorrect assumption that MPos == PPos
-			var heightOffset = Math.Min(Grid.MaximumTerrainHeight, MapSize.Y - Bounds.Bottom);
+			var top = int.MaxValue;
+			var bottom = int.MinValue;
+
+			if (Grid.MaximumTerrainHeight > 0)
+			{
+				// The minimap is drawn in cell space, so we need to
+				// unproject the PPos bounds to find the MPos boundaries.
+				// This matches the calculation in RadarWidget that is used ingame
+				for (var x = Bounds.Left; x < Bounds.Right; x++)
+				{
+					var allTop = Unproject(new PPos(x, Bounds.Top));
+					var allBottom = Unproject(new PPos(x, Bounds.Bottom));
+					if (allTop.Any())
+						top = Math.Min(top, allTop.MinBy(uv => uv.V).V);
+
+					if (allBottom.Any())
+						bottom = Math.Max(bottom, allBottom.MinBy(uv => uv.V).V);
```

Comment:
Doesn't this actually have to by `allBottom.MaxBy`? (If so https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Widgets/RadarWidget.cs#L193 is also wrong.)

Reply:
Without a descriptive comment explaining otherwise, I agree that this was mostly like a copy/paste error.

In practice, this doesn't actually change anything (I ran a test comparing old vs new values for all ts maps), so we can take the chance on changing it.

Reply:
If this doesn't change anything in practice, do we even need to MinBy or MaxBy? (Both here and above.)

Reply:
yeah because in principle allBottom *might* contain more than one element, even if it never does for the current set of maps.